jsinterop-ts-defs-processor/src/main/java/com/vertispan/tsdefs/builders/JavaToTsTypeConverter.java

```diff
@@ -212,8 +212,7 @@ public static boolean isSameType(
   }
 
   public static boolean isVoidType(TypeMirror typeMirror, HasProcessorEnv env) {
-    return (typeMirror.getKind().isPrimitive() && typeMirror.getKind() == TypeKind.VOID)
-        || isSameType(typeMirror, Void.class, env);
+    return isSameType(typeMirror, Void.class, env) || "void".equals(typeMirror.toString());
```

Comment:
This surprises me - and I believe in eclipse JDT that the toString will not return what you expect here, so you might want to do TypeName.get(...) to get a proper toString.

What is the typeMirror in the case of `void`, if not a primitive type?